src/formatDate.js

```diff
@@ -1,14 +1,38 @@
 'use strict';
 
-/**
+/*
  * @param {string} date
  * @param {string[]} fromFormat
  * @param {string[]} toFormat
  *
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const receivedDate = {};
+  const changedDate = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        receivedDate['YY'] = dateArr[i].slice(2);
+        receivedDate['YYYY'] = dateArr[i];
+        break;
+      case 'YY':
+        receivedDate['YYYY'] = `${dateArr[i] < 30 ? 20 : 19}${dateArr[i]}`;
+        receivedDate['YY'] = dateArr[i];
+        break;
+      default:
+        receivedDate[fromFormat[i]] = dateArr[i];
+        break;
+    }
+  }
+
+  for (const elem of toFormat.slice(0, -1)) {
+    changedDate.push(receivedDate[elem]);
```

Comment:
When pushing elements into `changedDate`, there's no check to ensure that `receivedDate[elem]` exists. This could lead to `undefined` values in the output if `toFormat` contains elements not present in `receivedDate`. It would be safer to check that `receivedDate[elem]` is not `undefined` before pushing it to `changedDate`.